diff --git a/src/crossbar_maintenance.erl b/src/crossbar_maintenance.erl
--- a/src/crossbar_maintenance.erl
+++ b/src/crossbar_maintenance.erl
@@ -180,5 +180,5 @@
           {'ok', kz_json:object()} | {'error', any()}.
 set_effective_autoload_modules(Modules) ->
-    case kapps_config:get_category(?CONFIG_CAT, 'false') of
+    case kz_datamgr:open_doc(?KZ_CONFIG_DB, ?CONFIG_CAT) of
         {'ok', Config} ->
             Node = kz_term:to_binary(node()),
