diff --git a/src/crossbar_auth.erl b/src/crossbar_auth.erl
--- a/src/crossbar_auth.erl
+++ b/src/crossbar_auth.erl
@@ -197,6 +197,7 @@ validate_auth_token(Token) ->
 validate_auth_token(Token, Options) ->
     case kz_auth:validate_token(Token, Options) of
         {'error', 'no_jwt_signed_token'} -> maybe_db_token(Token);
+        {'error', _Reason}=Error -> Error;
         {'ok', Claims} ->
             ProxyIPs = props:get_value(<<"proxy_ips">>, Options),
             maybe_check_proxy(Claims, ProxyIPs)
