diff --git a/kazoo_apps/src/kazoo_apps_init.erl b/kazoo_apps/src/kazoo_apps_init.erl
index 46e4e3c7d..01bb82bc1 100644
--- a/kazoo_apps/src/kazoo_apps_init.erl
+++ b/kazoo_apps/src/kazoo_apps_init.erl
@@ -43,7 +43,7 @@ set_dns_cache_options() ->
 -spec set_cookie() -> 'true'.
 set_cookie() ->
     Cookie = maybe_cookie_from_env(),
-    lager:info("setting ~s cookie to ~p", [?APP, Cookie]),
+    lager:info("setting Erlang cookie for ~s", [?APP]),
     erlang:set_cookie(node(), Cookie).
 
 -spec maybe_cookie_from_env() -> atom().
diff --git a/sup/src/sup.erl b/sup/src/sup.erl
index 2706bcb06..547fa8994 100644
--- a/sup/src/sup.erl
+++ b/sup/src/sup.erl
@@ -149,7 +149,7 @@ get_target(Options, Verbose) ->
     case net_adm:ping(Target) of
         'pong' ->
             Verbose
-                andalso stdout("Connected to service ~s with cookie ~s", [Target, Cookie]),
+                andalso stdout("Connected to service ~s", [Target]),
             Target;
         'pang' ->
             stderr("Connection to service failed!", []),
@@ -242,8 +242,8 @@ print_no_setcookie() ->
     halt(1).
 
 -spec print_ping_failed(string(), atom()) -> no_return().
-print_ping_failed(Target, Cookie) ->
-    stdout("Failed to connect to service ~s with cookie ~s", [Target, Cookie]),
+print_ping_failed(Target, _Cookie) ->
+    stdout("Failed to connect to service ~s", [Target]),
     stdout("  Possible fixes:", []),
     stdout("    * Ensure the Kazoo service you are trying to connect to is running on the host", []),
     stdout("    * Ensure that you are using the same cookie as the Kazoo node, `sup -c <cookie>`", []),
@@ -289,7 +289,7 @@ stderr(Format, Things) ->
 option_spec_list() ->
     [{'help', $?, "help", 'undefined', "Show the program options"}
     ,{'node', $n, "node", {'string', from_env("KAZOO_NODE", "kazoo_apps")}, "Node name"}
-    ,{'cookie', $c, "cookie", {'string', from_env("KAZOO_COOKIE", "change_me")}, "Erlang cookie"}
+    ,{'cookie', $c, "cookie", {'string', from_env("KAZOO_COOKIE", "")}, "Erlang cookie"}
     ,{'timeout', $t, "timeout", {'integer', 0}, "Command timeout"}
     ,{'verbose', $v, "verbose", 'undefined', "Be verbose"}
     ,{'module', 'undefined', 'undefined', 'string', "The name of the remote module"}
