diff --git a/kazoo_config.c b/kazoo_config.c
--- a/kazoo_config.c
+++ b/kazoo_config.c
@@ -221,8 +221,12 @@ switch_status_t kazoo_config_filters(switch_memory_pool_t *pool, switch_xml_t cfg
 			if(compare) {
 				if (!strncmp(compare, "value", 7)) {
 					cur->compare = FILTER_COMPARE_VALUE;
-				} else if (!strncmp(compare, "prefix", 6)) {
+				} else if (!strcmp(compare, "prefixes")) {
+						cur->compare = FILTER_COMPARE_PREFIXES;
+				} else if (!strcmp(compare, "prefix")) {
 						cur->compare = FILTER_COMPARE_PREFIX;
+				} else if (!strcmp(compare, "contains")) {
+						cur->compare = FILTER_COMPARE_CONTAINS;
 				} else if (!strncmp(compare, "list", 4)) {
 						cur->compare = FILTER_COMPARE_LIST;
 				} else if (!strncmp(compare, "exists", 6)) {
@@ -237,7 +241,7 @@ switch_status_t kazoo_config_filters(switch_memory_pool_t *pool, switch_xml_t cfg
 			if(cur->value == NULL)
 				cur->compare = FILTER_COMPARE_EXISTS;
 
-			if(cur->compare == FILTER_COMPARE_LIST) {
+			if(cur->compare == FILTER_COMPARE_LIST || cur->compare == FILTER_COMPARE_PREFIXES) {
 				cur->list.size = switch_separate_string(cur->value, '|', cur->list.value, MAX_LIST_FIELDS);
 			}
 
@@ -282,7 +282,9 @@ switch_status_t kazoo_config_field(kazoo_config_ptr definitions, switch_memory_p
 			cur->in_type = FIELD_FIRST_OF;
 		} else if (!strncmp(type, "expand", 6)) {
 			cur->in_type = FIELD_EXPAND;
-		} else if (!strncmp(type, "prefix", 10)) {
+		} else if (!strcmp(type, "prefixes")) {
+			cur->in_type = FIELD_PREFIXES;
+		} else if (!strcmp(type, "prefix")) {
 			cur->in_type = FIELD_PREFIX;
 		} else if (!strncmp(type, "group", 5)) {
 			cur->in_type = FIELD_GROUP;
@@ -329,7 +331,7 @@ switch_status_t kazoo_config_field(kazoo_config_ptr definitions, switch_memory_p
 		cur->out_type = JSON_STRING;
 	}
 
-	if(cur->in_type == FIELD_FIRST_OF) {
+	if(cur->in_type == FIELD_FIRST_OF || cur->in_type == FIELD_PREFIXES) {
 		cur->list.size = switch_separate_string(cur->value, '|', cur->list.value, MAX_LIST_FIELDS);
 	}
 
diff --git a/kazoo_fields.h b/kazoo_fields.h
--- a/kazoo_fields.h
+++ b/kazoo_fields.h
@@ -81,6 +81,8 @@ typedef enum {
 	FILTER_COMPARE_LIST,
 	FILTER_COMPARE_VALUE,
 	FILTER_COMPARE_PREFIX,
+	FILTER_COMPARE_PREFIXES,
+	FILTER_COMPARE_CONTAINS,
 	FILTER_COMPARE_EXISTS,
 	FILTER_COMPARE_FIELD
 
@@ -119,6 +119,7 @@ typedef enum {
 	FIELD_FIRST_OF,
 	FIELD_EXPAND,
 	FIELD_PREFIX,
+	FIELD_PREFIXES,
 	FIELD_OBJECT,
 	FIELD_GROUP,
 	FIELD_REFERENCE
diff --git a/kazoo_message.c b/kazoo_message.c
--- a/kazoo_message.c
+++ b/kazoo_message.c
@@ -86,6 +86,26 @@ static int inline filter_compare(switch_event_t* evt, kazoo_filter_ptr filter, ka
 		}
 		break;
 
+	case FILTER_COMPARE_PREFIXES:
+		for (header = evt->headers; header && !hasValue; header = header->next) {
+			for (n = 0; n < filter->list.size; n++) {
+				if (!strncmp(header->name, filter->list.value[n], strlen(filter->list.value[n]))) {
+					hasValue = 1;
+					break;
+				}
+			}
+		}
+		break;
+
+	case FILTER_COMPARE_CONTAINS:
+		if (*filter->name == '$') {
+			value = expr = kz_event_expand_headers(evt, filter->name);
+		} else {
+			value = switch_event_get_header(evt, filter->name);
+		}
+		hasValue = value && strstr(value, filter->value) ? 1 : 0;
+		break;
+
 	case FILTER_COMPARE_LIST:
 		if (*filter->name == '$') {
 			value = expr = kz_event_expand_headers(evt, filter->name);
@@ -309,6 +309,34 @@ cJSON * kazoo_event_add_field_to_json(cJSON *dst, switch_event_t *src, kazoo_fie
 
 			break;
 
+		case FIELD_PREFIXES:
+
+			for (header = src->headers; header; header = header->next) {
+				for (n = 0; n < field->list.size; n++) {
+					const char *prefix = field->list.value[n];
+					size_t prefix_len = strlen(prefix);
+
+					if (!strncmp(header->name, prefix, prefix_len)) {
+						const char *name = field->exclude_prefix ? header->name + prefix_len : header->name;
+
+						if (header->idx) {
+							cJSON *array = cJSON_CreateArray();
+
+							for (i = 0; i < header->idx; i++) {
+								cJSON_AddItemToArray(array, kazoo_event_json_value(field->out_type, header->array[i]));
+							}
+
+							kazoo_cJSON_AddItemToObject(dst, name, array);
+						} else {
+							kazoo_event_add_json_value(dst, field, name, header->value);
+						}
+						break;
+					}
+				}
+			}
+
+			break;
+
 		case FIELD_STATIC:
 			item = kazoo_event_add_json_value(dst, field, field->name, field->value);
 			break;
