diff --git a/kazoo_fetch_agent.c b/kazoo_fetch_agent.c
index e91b2e4..e367bd4 100644
--- a/kazoo_fetch_agent.c
+++ b/kazoo_fetch_agent.c
@@ -85,7 +85,7 @@ static char *expand_vars(char *xml_str) {
 				switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "trying to expand %s \n", var);
 				if ((val = switch_core_get_variable_dup(var))) {
 					char *p;
-					switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "expanded %s to %s\n", var, val);
+					switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "expanded variable %s\n", var);
 					for (p = val; p && *p && wp <= ep; p++) {
 						*wp++ = *p;
 					}
@@ -315,11 +315,10 @@ static switch_xml_t fetch_handler(const char *section, const char *tag_name, con
 		if (kazoo_globals.expand_headers_on_fetch) {
 			reply.xml_str = expand_vars(reply.xml_str);
 		}
-		switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "Received %s XML (%s) after %dms: %s\n"
+		switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "Received %s XML (%s) after %dms\n"
 						  ,section
 						  ,reply.uuid_str
-						  ,(unsigned int) (switch_micro_time_now() - now) / 1000
-						  ,reply.xml_str);
+						  ,(unsigned int) (switch_micro_time_now() - now) / 1000);
 		if ((xml = switch_xml_parse_str_dynamic(reply.xml_str, SWITCH_FALSE)) == NULL) {
 			switch_safe_free(reply.xml_str);
 		}
